mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRV.cpp

```diff
@@ -591,10 +592,12 @@ class GPUSubgroupReduceConversion final
   LogicalResult
   matchAndRewrite(gpu::SubgroupReduceOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    auto opType = op.getOp();
-    auto result =
-        createGroupReduceOp(rewriter, op.getLoc(), adaptor.getValue(), opType,
-                            /*isGroup*/ false, op.getUniform());
+    if (isa<VectorType>(adaptor.getValue().getType()))
```

Comment:
IMO, instead of rejecting vector, better to check types which are supported:
```
if (!isa<IntegerType, FloatType>(...))
```
Also, `notifyMatchFailure`

Reply:
Done, thanks